scripts/ci/enforce_runtime_determinism.py

```
@@ -0,0 +1,196 @@
+#!/usr/bin/env python3
+"""
+Fail-closed determinism enforcement for backend runtime code.
+"""
+from __future__ import annotations
+
+import argparse
+import ast
+from pathlib import Path
+from typing import Iterable, List, Optional, Set
```

Ruff (0.14.11)

Import from `collections.abc` instead: `Iterable`

Import from `collections.abc`

(UP035)